### PR TITLE
Rejudge single submission

### PR DESCRIPTION
**Pull request by henryyang42**

### CODE REVIEW

File: status/views.py [hunk 1]
```diff
@@ -155,3 +158,16 @@ def view_code(request, sid):
...
         logger.warning('File %s Not Found!' % filename)
         raise Http404('File %s Not Found!' % filename)
+
+
+@login_required()
+def rejudge(request, sid):
+    submission = get_object_or_404(Submission, id=sid)
+    if can_rejudge(submission, request.user):
+        rejudge_submission(submission)
+        print request.GET
```
Comment (nickelcheng) [inline status/views.py]: Please remove this and add log

File: status/views.py [hunk 2]
```diff
@@ -155,3 +158,16 @@ def view_code(request, sid):
...
+        messages.success(request, 'Submission %s rejuded' % sid)
+        return redirect('%s?%s' % \
+            (reverse('status:status'), urllib.urlencode(request.GET)))
```
Comment (nickelcheng) [inline status/views.py]: what's the usage of urlencode?

File: status/templatetags/status_filters.py [hunk 3]
```diff
@@ -99,6 +100,38 @@ def show_detail(submission, user):
     return False
 
 
+@register.filter()
+def can_rejudge(submission, user):
```
Comment (nickelcheng) [inline status/templatetags/status_filters.py]: The rule seems to be perfect.

File: status/templates/status/statusTable.html [hunk 4]
```diff
@@ -34,6 +34,11 @@
           {% else %}
             {{ submission.grouper.id }}
           {% endif %}
+          {% if submission.grouper|can_rejudge:user %}
+            <a href="{% url "status:rejudge" submission.grouper.id %}?{{ request.GET.urlencode }}">
```
Comment (nickelcheng) [inline status/templates/status/statusTable.html]: Can you give an confirm before rejudge?

File: status/views.py (hunk 2 above)
Comment (henryyang42) [inline status/views.py]: Make status remain in the same page.
We use get parameter  ``?username=&pid=&cid=&status=`` to do that job.

File: status/templates/status/statusTable.html (hunk 4 above)
Comment (henryyang42) [inline status/templates/status/statusTable.html]: Judging single submission with alert seems to be annoying.

**Comment by bruce3557** [comment]
Wonder know what situation we need to use this function but not rejudge all the submissions of a problem

**Comment by nickelcheng** [comment]
Judge error, or 魯小小student

**Comment by henryyang42** [comment]
What is  ``魯小小student``? XD

**Comment by nickelcheng** [comment]
It often occurs in tests.
A student that think his/her solution is correct, but our judge give it not accepted.
After the test, the student will ask TA to rejudge some of his/her submissions, then get the same result, finally give up reluctantly.

**Comment by bruce3557** [comment]
Zz

**Comment by bruce3557** [comment]
Fine

**Comment by nickelcheng** [comment]
+1, @CHAHSINHOU, please test

**Comment by nickelcheng** [comment]
test ok, going to merge.

**Pull request merged**